administrative-sdk/connection.js

```diff
@@ -281,35 +264,27 @@ class Connection {
    * @param {string} [auth] The authorization header value to pass along with the request.
    */
   _ajaxDelete(url, auth) {
-    var request = new XMLHttpRequest();
-    var response = null;
-    request.open('DELETE', url);
+    const headers = new Headers();
     if (typeof auth !== 'undefined') {
-      request.setRequestHeader('Authorization', auth);
+      headers.append('Authorization', auth);
     }
-    return new Promise(function(resolve, reject) {
-      request.onload = function() {
-        if (request.status >= 100 && request.status < 300) {
-          // A delete call usually has no body to parse.
-          resolve();
-        } else {
-          // Some error occured.
-          try {
-            response = Connection._parseResponse(request.responseText);
-            reject(
-              {
-                errors: response.errors,
-                status: request.status
-              }
-            );
-          } catch (e) {
-            reject(response || e);
-          }
-        }
-      };
-
-      request.send();
-    });
+    const options = {
+      method: 'DELETE',
+      headers: headers
+    };
+    return fetch(url, options)
+      .then(response =>
+        (response.json()
+          .then(data => {
+            if (response.ok) {
+              return data;
+            }
+            throw data;
+          })
+      ))
+      .catch(error => {
+        throw error;
+      });
```

Comment:
Same as in `_ajaxGet`. Although I do think some code could be shared between those two.